### PR TITLE
Add Security Linting to CI pipeline and ensure that linters can produce output (mat-1167)

### PR DESCRIPTION
**Pull request by temaniarpit27**

### CODE REVIEW

**Comment by temaniarpit27** [comment]
@jdkanani : how to remove circleci check from heimdall repo?

**Comment by temaniarpit27** [comment]
arpit@arpit-ThinkPad-T480:~/go/src/github.com/maticnetwork/heimdall/auth$ golangci-lint run -E gosec --skip-dirs-use-default --skip-files ".*_test.go$"
types/stdtx.go:138:9: SA1019: common.ToHex is deprecated: use hexutil.Encode instead.  (staticcheck)
	return common.ToHex(ss)
	       ^



arpit@arpit-ThinkPad-T480:~/go/src/github.com/maticnetwork/heimdall/types$ golangci-lint run -E gosec --skip-dirs-use-default --skip-files ".*_test.go$"
rest/rest.go:202:9: S1034: assigning the result of this type assertion to a variable (switch resp := resp.(type)) could eliminate the following type assertions:
	rest/rest.go:204:12 (gosimple)
	switch resp.(type) {
	       ^
bytes.go:80:9: SA1019: common.ToHex is deprecated: use hexutil.Encode instead.  (staticcheck)
	return common.ToHex(bz)
	       ^

**Comment by temaniarpit27** [comment]
Errors:
Not sure how to resolve them @jdkanani @mankenavenkatesh 
```
 arpit@arpit-ThinkPad-T480:~/go/src/github.com/maticnetwork/heimdall/helper$ golangci-lint run -E gosec --skip-dirs-use-default --skip-files ".*_test.go$"
util.go:705:15: G107: Potential HTTP request made with variable url (gosec)
	resp, err := http.Get(URL)
	             ^
```
```
 arpit@arpit-ThinkPad-T480:~/go/src/github.com/maticnetwork/heimdall/staking$ golangci-lint run -E gosec --skip-dirs-use-default --skip-files ".*_test.go$"
querier.go:270:43: SA1006: printf-style function with dynamic format string and no further arguments should use print-style function instead (staticcheck)
		return nil, sdk.ErrInternal(fmt.Sprintf(err.Error()))
		                                        ^
```

```
 arpit@arpit-ThinkPad-T480:~/go/src/github.com/maticnetwork/heimdall/types$ golangci-lint run -E gosec --skip-dirs-use-default --skip-files ".*_test.go$"
rest/rest.go:202:9: S1034: assigning the result of this type assertion to a variable (switch resp := resp.(type)) could eliminate the following type assertions:
	rest/rest.go:204:12 (gosimple)
	switch resp.(type) {
	       ^
bytes.go:80:9: SA1019: common.ToHex is deprecated: use hexutil.Encode instead.  (staticcheck)
	return common.ToHex(bz)
	       ^
```

**Comment by temaniarpit27** [comment]
@jdkanani : please stop circle ci check from dashboard before this PR gets merged
@jdkanani @mankenavenkatesh : there were some issues in test files. even after running the golang-ci lint command and skipping files, it wasn't working properly. for now i have tested by removing those files from the folder and tested it
i havent enabled lint with the `make build` because of this reason.
i would say 99% of the issues are fixed. you guys can go ahead and review it once.

File: checkpoint/handler.go [hunk 1]
```diff
@@ -95,7 +95,11 @@ func handleMsgCheckpoint(ctx sdk.Context, msg types.MsgCheckpoint, k Keeper, con
 	// Calculate new account root hash
 	dividendAccounts := k.sk.GetAllDividendAccounts(ctx)
 	k.Logger(ctx).Debug("DividendAccounts of all validators", "dividendAccounts", dividendAccounts)
-	accountRoot, err := types.GetAccountRootHash(dividendAccounts)
+	var accountRoot []byte
+	accountRoot, err = types.GetAccountRootHash(dividendAccounts)
+	if err != nil {
+		k.Logger(ctx).Error("handleMsgCheckpoint | GetAccountRootHash", "Error", err)
```
Comment (jdkanani) [inline checkpoint/handler.go:101]: We should return an error if account hash an error

File: checkpoint/handler.go [hunk 2]
```diff
@@ -117,14 +121,16 @@ func handleMsgCheckpoint(ctx sdk.Context, msg types.MsgCheckpoint, k Keeper, con
...
+	if err := k.SetCheckpointBuffer(ctx, hmTypes.CheckpointBlockHeader{
 		StartBlock:      msg.StartBlock,
 		EndBlock:        msg.EndBlock,
 		RootHash:        msg.RootHash,
 		AccountRootHash: msg.AccountRootHash,
 		Proposer:        msg.Proposer,
 		TimeStamp:       timeStamp,
-	})
+	}); err != nil {
+		k.Logger(ctx).Error("handleMsgCheckpoint | SetCheckpointBuffer", "Error", err)
```
Comment (jdkanani) [inline checkpoint/handler.go:132]: Same! Return error

File: checkpoint/handler.go [hunk 3]
```diff
@@ -198,7 +204,9 @@ func handleMsgCheckpointAck(ctx sdk.Context, msg types.MsgCheckpointAck, k Keepe
 	}
 
 	// Add checkpoint to headerBlocks
-	k.AddCheckpoint(ctx, msg.HeaderBlock, *headerBlock)
+	if err := k.AddCheckpoint(ctx, msg.HeaderBlock, *headerBlock); err != nil {
+		k.Logger(ctx).Error("handleMsgCheckpointAck | AddCheckpoint", "error", err)
```
Comment (jdkanani) [inline checkpoint/handler.go:208]: Return error

File: clerk/keeper.go [hunk 4]
```diff
@@ -182,7 +182,9 @@ func (k *Keeper) IterateRecordsAndApplyFn(ctx sdk.Context, f func(record types.E
 	for ; iterator.Valid(); iterator.Next() {
 		// unmarshall span
 		var result types.EventRecord
-		k.cdc.UnmarshalBinaryBare(iterator.Value(), &result)
+		if err := k.cdc.UnmarshalBinaryBare(iterator.Value(), &result); err != nil {
+			k.Logger(ctx).Error("IterateRecordsAndApplyFn | UnmarshalBinaryBare", "error", err)
```
Comment (jdkanani) [inline clerk/keeper.go:186]: Should return

File: client/tx/encode.go [hunk 5]
```diff
@@ -86,6 +86,10 @@ If you supply a dash (-) argument in place of an input filename, the command rea
 			}
 
 			txBytes, err := helper.GetStdTxBytes(cliCtx, stdTx)
+			if err != nil {
+				return
```
Comment (jdkanani) [inline client/tx/encode.go:90]: Return err

**Comment by temaniarpit27** [comment]
@jdkanani fixed review comments

**Pull request merged**